engine/src/main/java/org/hibernate/validator/constraints/ISBN.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.constraints;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.ReportAsSingleViolation;
+
+import org.hibernate.validator.constraints.ISBN.List;
+
+/**
+ * Checks that the annotated character sequence is a valid
+ * <a href="https://en.wikipedia.org/wiki/International_Standard_Book_Number">ISBN</a> number.
+ * The length of the number and the check digit are both verified.
```

Comment:
Do we also support contain dashes (as with the other check numbers)? If so, it'd be worth stating that here.